LEAF_Request_Portal/sources/FormEditor.php

```diff
@@ -261,12 +261,17 @@ function setSensitive($indicatorID, $input)
     function setDisabled($indicatorID, $input) 
     {
     	$disabledTime = 0;
-    	if($input >= 1) {
+    	if($input == 1) {
+            $this->setRequired($indicatorID, 0);
+            $this->disableSubindicators($indicatorID);
+    	    $disabledTime = 1;
+    	}
+    	elseif ($input == 2){
     		$this->setRequired($indicatorID, 0);
     		$this->disableSubindicators($indicatorID);
     		$disabledTime = time();
     	}
-    	
+

```

Comment:
Without looking too deep at this function and what it does, I would say this needs an "else" clause to make sure the object variables setRequired and disabledSubIndicators are at default values.  Not for sure, but if you have an "elseif" then you usually want to have final "else" clause to cover users being.... "creative".